### PR TITLE
Use setuptools_scm for version number

### PR DESCRIPTION
**Pull request by kotfu**
For issue #470, remove hard coded version numbers. Instead use setuptools_scm, which derives the version number from git tags.

`setuptools_scm` has a couple of drawbacks:

- you can't set the version number in any other way than by a git tag. If you want to increment the version number without tagging, you are out of luck.
- `setuptools_scm` is opinionated. If you follow semantic versioning, you must tag minor releases with the a patch level of '0', i.e. '2.0.0', not '2.0'
- if HEAD is not pointing at a tag, you get a long version number, i.e. `0.9.4.dev73+g4da2d37.d20180729`. This is only visible to developers (because only developers should be running untagged versions of `cmd2`), but can be modified if we really want to change it.

The good news is that this PR takes away much of the pain of version increments.

### CODE REVIEW

**Comment by kotfu** [comment]
This PR is currently failing on the VSTS build. I think it might be a problem with the build configuration. The test that is failing does:
`
assert cmd2.__version__
`

The only way that `__version__` can generates an `AttributeError` (which is what is happening) is if the `cmd2` package is not installed. I think that on the VSTS build, it's testing the source code in place, not installing the `cmd2` package and testing the installed package.

**Comment by tleonhardt** [comment]
@kotfu I configured the VSTS macOS build to simply use pytest directly, so yeah it is just testing the code in place.  My bad.

In the long run I’m going to switch the VSTS build to be Yaml based so the configuration is in our GitHub repo, but that feature is in preview so I didn’t want to rely on it yet.

In the meantime, see if you can create a free account at visualstudio.com and send me an email with your account name and I’ll add you as an admin to the cmd2 project there so you can tweak the build configuration.

File: cmd2/__init__.py [hunk 1]
```diff
@@ -1,6 +1,14 @@
 #
 # -*- coding: utf-8 -*-
 """This simply imports certain things for backwards compatibility."""
-from .cmd2 import __version__, Cmd, Statement, EmptyStatement, categorize
+
+from pkg_resources import get_distribution, DistributionNotFound
+try:
+    __version__ = get_distribution(__name__).version
```
Comment (tleonhardt) [inline cmd2/__init__.py]: So this is finally going to make me abandon my PYTHONPATH hack way of doing ``cmd2`` development and switch to the **pip** way instead using:
```sh
$ cd ~/src/cmd2
$ pip install -e .
```

But that is probably a good thing ...

**Comment by tleonhardt** [comment]
@kotfu I modified the VSTS build so that it first uses **pip** to install ``cmd2`` via:
```sh
$ pip install -e .
```

You should have gotten an email invite to join that VSTS account/project.  

The VSTS macOS build capability is in somewhat of a preview state.  A little down the road we will switch to using a YAML file to define the build to make it obvious what is going on and we will probably test on a couple versions of Python on macOS instead of just one.  But I figured some automated build on macOS is better than no automated build on macOS ...

**Pull request merged**